src/hotspot/share/runtime/threadWXSetters.inline.hpp

```diff
@@ -28,25 +28,60 @@
 
 // No threadWXSetters.hpp
 
-#if defined(__APPLE__) && defined(AARCH64)
+#ifdef MACOS_W_XOR_X
 
+#include "classfile/classLoader.hpp"
+#include "runtime/perfData.inline.hpp"
 #include "runtime/thread.inline.hpp"
 
 class ThreadWXEnable  {
-  Thread* _thread;
+  Thread *_thread;
```

Comment:
* needs to be attached to type not field here and below.

Reply:
> * needs to be attached to type not field here and below.

I'm afraid this is against the guidelines. At least that is what I've been told, but I can't find it in the guidelines, so maybe that isn't true.

Reply:
No, it's not explicitly stated in the [guidelines](https://github.com/openjdk/jdk/blob/master/doc/hotspot-style.md) but it is adopted 'almost everywhere' in the code base. Likewise with reference parameter declarations the '&' attaches to the type.

Reply:
Ah, I see. I mistakenly thought the snippet above was a suggestion you were making...